cumulusci/tasks/bulkdata/generate_from_yaml.py

```diff
@@ -137,36 +134,38 @@ def open_new_continuation_file(self) -> Optional[TextIO]:
             new_continuation_file = None
         return new_continuation_file
 
-    def generate_data(self, session, engine, base, num_records, current_batch_num):
-        output_stream = SqlOutputStream(engine, self.mappings)
+    def generate_data(self, db_url, num_records, current_batch_num):
+        output_stream = SqlOutputStream.from_url(db_url, self.mappings)
         old_continuation_file = self.get_old_continuation_file()
         if old_continuation_file:
             # reopen to ensure file pointer is at starting point
             old_continuation_file = open(old_continuation_file, "r")
         new_continuation_file = self.open_new_continuation_file()
 
-        with open(self.yaml_file) as open_yaml_file:
-            summary = generate(
-                open_yaml_file=open_yaml_file,
-                user_options=self.vars,
-                output_stream=output_stream,
-                stopping_criteria=self.stopping_criteria,
-                continuation_file=old_continuation_file,
-                generate_continuation_file=new_continuation_file,
-            )
+        try:
+            with open(self.yaml_file) as open_yaml_file:
+                summary = generate(
+                    open_yaml_file=open_yaml_file,
+                    user_options=self.vars,
+                    output_stream=output_stream,
+                    stopping_criteria=self.stopping_criteria,
+                    continuation_file=old_continuation_file,
+                    generate_continuation_file=new_continuation_file,
+                )
+        finally:
             output_stream.close()
```

Comment:
Maybe output streams should be context managers.

Reply:
I can do that in Snowfakery but I'd rather not upgrade them both in lockstep this time. I can do it in Snowfakery and then clean this up next time I'm in this code.

Reply:
Yeah, I meant it that way rather than a blocker to merge this.